docs/basics/101-123-yoda.rst

```diff
@@ -0,0 +1,433 @@
+  one data component for each of them. And if your analysis
+  relies on two or more data collections, these collections should each be an
+  individual component, not combined into one.
+
+- Store scripts or **code** used for the analysis of data in a dedicated ``code/``
+  directory, outside of the data component of the dataset.
+
+- Collect **results** of an analysis in a dedicated ``outputs/`` directory, and
+  leave the input data of an analysis untouched by your computations.
+
+- Include a place for complete **execution environments**, for example
+  `singularity images <https://singularity.lbl.gov/>`_ or
+  `docker containers <https://www.docker.com/get-started>`_ [#f2]_, in
+  the form of an ``envs/`` directory, if relevant for your analysis.
+
+- And if you conduct multiple different analyses, create a dedicated
+  project for each analysis, instead of conflating them.
+
+This, for example, would be a directory structure of a very comprehensive [#f3]_
+data analysis project complying to the YODA principles:
+
+.. code-block:: bash
+
+    ├── ci/                         # continuous integration configuration
+    │   └── .travis.yml
+    ├── code/                       # your code
+    │   ├── tests/                  # unit tests to test your code
+    │   │   └── test_myscript.py
+    │   └── myscript.py
+    ├── docs                        # documentation about the project
+    │   ├── build/
+    │   └── source/
+    ├── envs                        # computational environments
+    │   └── Singularity
+    ├── inputs/                     # dedicated inputs/, will not be changed by an analysis
+    │   └─── data/
+    │       ├── dataset1/           # one stand-alone data component
+    │       │   └── datafile_a
+    │       └── dataset2/
+    │           └── datafile_a
+    ├── outputs/                    # outputs away from the input data
+    │   └── important_results/
+    │       └── figures/
+    ├── CHANGELOG.md                # notes for fellow humans about your project
+    ├── HOWTO.md
+    └── README.md
+
+
+There are many advantages to this modular way of organizing contents.
+Having input data as independent components that are not influenced (only
+consumed) by an analysis does not conflate the data for
+an analysis with the results or the code, thus assisting understanding
+the project for anyone unfamiliar with it.
+But more than just structure, this organization aids modular reuse or
+publication of the individual components, for example data.
+Later you might also write a scientific paper about your analysis.
+This whole dataset can easily become a modular component in a paper
+project, to make sharing paper, code, data, and results easy.
+The usecase :ref:`remodnav` contains a step-by-step instruction on
+how to build and share such a reproducible paper, if you want to learn
+more.
+
+.. figure:: ../img/dataset_modules.svg
+   :figwidth: 100%
+   :alt: Modular structure of a data analysis project
```

Comment:
It would be great to link the directory tree shown above to this figure. They are essentially the same thing, but one emphasizes a "compressed" structure of all the individual pieces on the file system from the point of view of a single top-level dataset with a particular purpose. The figure instead shows more of a process view on a project.